server/executor/runner.go

```diff
@@ -152,6 +160,14 @@ func (r persistentRunner) Run(ctx context.Context, testObj test.Test, metadata t
 
 	r.listenForStopRequests(ctx, cancelCtx, run)
 
+	// configuring required gates
+	if requiredGates != nil {
```

Comment:
Required gates can be overwritten by the clients when executing a test run and if it's not part of the request we fallback to the global configuration